src/dscanner/analysis/config.d

```diff
@@ -209,6 +209,12 @@ struct StaticAnalysisConfig
 	@INI("Check for unused function return values")
 	string unused_result = Check.enabled;
 
+	@INI("Enable cyclomatic complexity check")
+	string cyclomatic_complexity = Check.enabled;
+
+	@INI("Maximum cyclomatic complexity after which to issue warnings")
+	int max_cyclomatic_complexity = 50;
```

Comment:
I wonder if there should be a separate class of warnings to distinguish "you might have a bug here" from "you need to change your code due to our team's policy / to make sure other people can easily understand/maintain it".

Reply:
One way to think of this is that the first class is useful to everyone, but the second class is useful only for people working in teams. If you are working on a personal or one-shot throwaway project, this class of warnings would be completely useless. So I'm actually wondering if it should perhaps be disabled by default.

Reply:
> One way to think of this is that the first class is useful to everyone, but the second class is useful only for people working in teams. If you are working on a personal or one-shot throwaway project, this class of warnings would be completely useless. So I'm actually wondering if it should perhaps be disabled by default.

Even for a one-off, I would prefer to have that warning exist even if it's quite high just to ensure I don't start having bad practices.